NEWS.md

```diff
@@ -1,5 +1,22 @@
 # vctrs (development version)
 
+* The `[<-` and `$<-` methods for `<list_of>` no longer shorten the list when assigning `NULL` into the list. This is consistent with the existing `[[<-` method for `<list_of>`.
+
+* `as_list_of()` on an existing `<list_of>` no longer has a `.ptype` argument for changing the type on the fly, as this feels incompatible with the new system that allows restricting both the type and size. If you really need this, coerce to a bare list with `as.list()` first, then coerce back to a `<list_of>` using the `<list>` method of `as_list_of()`.
```

Comment:
Does this have broader implications? Like if you have a date-time, what generic do you use to change the timezone? i.e. do we have a general philosophy of how you change one attribute while preserving the others?